{"id":"presto-1001","summary":"PrestoException Detected","description":"This is a generic Presto error, and there are many reasons this exception can occur. Check the following log message to understand the root cause.","keywords":["com.facebook.presto.spi.PrestoException"],"knowledge_center_links":[""]}
{"id":"presto-1002","summary":"ResultsException Detected","description":"Exception is related to the directory structure expected by Presto for bucketed tables.","keywords":["com.facebook.presto.jdbc.PrestoResultSet.resultsException"],"knowledge_center_links":[""]}
{"id":"presto-1003","summary":"Presto PageTransportTimeoutException Detected","description":"This exception is primarily for big queries which deal with a huge amount of data in join statements.","keywords":["com.facebook.presto.operator.PageTransportTimeoutException"],"knowledge_center_links":[""]}
{"id":"presto-1004","summary":"PrestoException Detected When Connecting to Trino Using Custom SQL","description":"This is a generic Presto error, and there are many reasons this exception can occur. Check the following log message to understand the root cause.","keywords":["io.prestosql.spi.PrestoException"],"knowledge_center_links":[""]}
{"id":"presto-1005","summary":"ParsingException Detected Related to Semantics While Writing Presto Query","description":"This exception occurs if the query (SQL) is not delimited properly or if there is any typo while running the Presto query, mostly occurring due to semicolons, commas, etc.","keywords":["com.facebook.presto.sql.parser.ParsingException"],"knowledge_center_links":[""]}
{"id":"presto-1006","summary":"Related to Load or Network Issue","description":"Presto worker nodes might be having resource contention on CPU or memory, which ultimately causes an issue while reading, writing, fetching data from other nodes, or overall Presto communication.","keywords":["com.facebook.presto.spi.PrestoTransportException"],"knowledge_center_links":[""]}
{"id":"presto-1007","summary":"Exceptions Related to Datatype Mismatch","description":"Implementation of the ResultSet class which is related to the datatype used (e.g., VARCHAR). When the value cannot be converted, we get this kind of exception.","keywords":["io.prestosql.jdbc.AbstractPrestoResultSet.resultsException"],"knowledge_center_links":[""]}
{"id":"presto-1008","summary":"More Memory Is Being Used","description":"This exception occurs when a Presto query is running and demands more memory at different stages, e.g., due to data skew or a task-concurrency factor.","keywords":["com.facebook.presto.ExceededMemoryLimitException"],"knowledge_center_links":[""]}
{"id":"presto-1009","summary":"Not a Valid Parquet File","description":"This exception is encountered when there is an issue with the Parquet file structure.","keywords": ["com.facebook.presto.parquet.ParquetCorruptionException"],"knowledge_center_links":[""]}
{"id":"presto-1010","summary":"No Permissions on Presto Tables","description":"This exception occurs mostly due to insufficient permissions or while performing some action on Hive tables as the Presto user.","keywords":["com.facebook.presto.spi.security.AccessDeniedException"],"knowledge_center_links":[""]}
{"id":"presto-1011","summary":"Mismatch in SQL Parameters","description":"This exception is mostly due to column mismatch, parameter mismatch, or datatype mismatch in the query.","keywords":["com.facebook.presto.sql.analyzer.SemanticException"],"knowledge_center_links":[""]}
{"id":"presto-1012","summary":"Table Does Not Exist","description":"When there is a table name mismatch.","keywords":["com.facebook.presto.spi.TableNotFoundException"],"knowledge_center_links":[""]}
{"id":"presto-1013","summary":"Access Denied or Permission Issue","description":"This exception is mostly due to table permissions set by the table owner or group, e.g., column or database/table name.","keywords":["io.prestosql.spi.security.AccessDeniedException"],"knowledge_center_links":[""]}
{"id":"presto-1014","summary":"No Connection to Hive Metastore","description":"When Presto loses connection to or is not able to connect to the Hive Metastore, we get this exception.","keywords":["com.facebook.presto.hive.metastore.thrift.Transport.rewriteException"],"knowledge_center_links":[""]}
{"id":"presto-1015","summary":"Due to Corrupt Statistics","description":"Fail queries reading Parquet files if statistics in those Parquet files are corrupt (e.g., min > max). To disable this behavior, set the configuration property 'hive.parquet.fail-on-corrupted-statistics' or session property 'parquet_fail_with_corrupted_statistics' to false.","keywords":["com.facebook.presto.parquet.predicate.TupleDomainParquetPredicate.failWithCorruptionException"],"knowledge_center_links":[""]}
{"id":"presto-1016","summary":"While Fetching Response from Worker Node","description":"This exception is mostly due to a long-running query when it fails to send or receive internal requests because of connection issues.","keywords":["com.facebook.presto.jdbc.internal.client.StatementClientV1.requestFailedException"],"knowledge_center_links":[""]}
{"id":"presto-1017","summary":"Access Denied Exception or Permission Issue","description":"Mostly related to how users or groups can access the Presto catalog.","keywords":["io.prestosql.spi.security.AccessDeniedException.denyCatalogAccess"],"knowledge_center_links":["https://issues.apache.org/jira/browse/RANGER-3142?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17468378#comment-17468378"]}
{"id":"presto-1018","summary":"Connection Being Refused by Presto Worker Node","description":"This exception is thrown while handling a JSON response, mostly due to some configuration mismatch or invalid response. Check any custom class implemented by the user.","keywords":["com.facebook.presto.server.smile.AdaptingJsonResponseHandler.handleException"],"knowledge_center_links":[""]}
{"id":"presto-1019","summary":"When Query Exceeds Per-Node Memory","description":"This is the exception when the memory per node exceeds more than what it is set to.","keywords":["io.prestosql.ExceededMemoryLimitException"],"knowledge_center_links":[""]}
{"id":"presto-1020","summary":"Access Table Content","description":"This exception occurs mostly due to table operations that are not permitted on Hive tables by Presto, possibly due to user or group-level permissions.","keywords":["com.facebook.presto.spi.security.AccessDeniedException.denySelectTable"],"knowledge_center_links":["https://github.com/prestodb/presto/issues/7773"]}
{"id":"presto-1021","summary":"Connection Issue - Presto JVM on Primary Instance","description":"This exception occurs when the Presto coordinator refuses connections from clients, mostly due to being overloaded. Make sure that the Presto JVM has enough memory to serve incoming connections.","keywords":["com.facebook.presto.jdbc.internal.airlift.http.client.RuntimeIOException"],"knowledge_center_links":[""]}
{"id":"presto-1022","summary":"When File Is Not Present in Amazon S3","description":"This exception occurs when Amazon S3 metadata is not synced or when EMRFS consistent view is disabled.","keywords":["com.facebook.presto.hive.s3.PrestoS3FileSystem.UnrecoverableS3OperationException"],"knowledge_center_links":[""]}
{"id":"presto-1023","summary":"Remote Service Is Not Available","description":"This is probably a transient issue or exception when Presto is not able to talk to some remote service due to a network issue. Or this can be a classic load on the JVM issue.","keywords":["com.facebook.presto.server.remotetask.SimpleHttpResponseHandler.ServiceUnavailableException"],"knowledge_center_links":[""]}